src/system.vala

```diff
@@ -4,6 +4,19 @@ namespace Frida {
 		public extern static Frida.HostApplicationInfo[] enumerate_applications ();
 		public extern static Frida.HostProcessInfo[] enumerate_processes ();
 		public extern static void kill (uint pid);
+
+#if LINUX
+		// memfd_create(2)
+		[CCode (cprefix = "MFD_", has_type_id = false, cheader_filename = "sys/mman.h")]
+		public enum MemfdFlags {
+			CLOEXEC,
+			ALLOW_SEALING,
+			HUGETLB
+		}
+
+		[CCode (cheader_filename = "sys/mman.h", cname="memfd_create")]
+		public extern static int memfd_create (string name, MemfdFlags flags);
```

Comment:
These are low-level helpers only used here, so I would move them to the bottom of the file and make them `private`. (Or later `internal` if needed elsewhere.) Actually if we move them to an internal namespace named `Linux`, it will be easy to update later when `linux.vapi` has these definitions.)